plugins/azure/fix_plugin_azure/resource/sql.py

```diff
@@ -0,0 +1,1293 @@
+from datetime import datetime
+from typing import Any, ClassVar, Dict, Optional, List, Type
+
+from attr import define, field
+
+from fix_plugin_azure.azure_client import AzureResourceSpec
+from fix_plugin_azure.resource.base import AzureSku, GraphBuilder, MicrosoftResource
+from fix_plugin_azure.resource.network import AzureSubnet
+from fixlib.baseresources import EdgeType, ModelReference
+from fixlib.json_bender import Bender, S, ForallBend, Bend
+from fixlib.types import Json
+
+service_name = "azure_sql"
+
+
+class AzureUserIdentity:
+    kind: ClassVar[str] = "azure_user_identity"
+    mapping: ClassVar[Dict[str, Bender]] = {"client_id": S("clientId"), "principal_id": S("principalId")}
+    client_id: Optional[str] = field(default=None, metadata={"description": "The Azure Active Directory client id."})
+    principal_id: Optional[str] = field(default=None, metadata={'description': 'The Azure Active Directory principal id.'})  # fmt: skip
```

Comment:
create a deferred edge to ms graph principal.
We should think about introducing something like `microsoft_graph_principal` which is one of :
- user
- group
- service principal
- managed identity